src/api/user/StatusAPI.js

```diff
@@ -0,0 +1,289 @@
+            return provider.getStatusForRole(role);
+        } else {
+            this.#userAPI.error("User provider does not support role status");
+        }
+    }
+
+    /**
+     * @param {import("./UserAPI").Role} role
+     * @returns {Promise<Boolean>} true if the configured UserProvider can provide status for the given role
+     * @see StatusUserProvider
+     */
+    canProvideStatusForRole(role) {
+        const provider = this.#userAPI.getProvider();
+
+        if (provider.canProvideStatusForRole) {
+            return provider.canProvideStatusForRole(role);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @param {import("./UserAPI").Role} role The role to set the status for.
+     * @param {Status} status The status to set for the provided role
+     * @returns {Promise<Boolean>} true if operation was successful, otherwise false.
+     */
+    setStatusForRole(role, status) {
+        const provider = this.#userAPI.getProvider();
+
+        if (provider.setStatusForRole) {
+            return provider.setStatusForRole(role, status);
+        } else {
+            this.#userAPI.error("User provider does not support setting role status");
+        }
+    }
+
+    /**
+     * Resets the status of the provided role back to its default status.
+     * @param {import("./UserAPI").Role} role The role to set the status for.
+     * @returns {Promise<Boolean>} true if operation was successful, otherwise false.
+     */
+    async resetStatusForRole(role) {
+        const provider = this.#userAPI.getProvider();
+        const defaultStatus = await this.getDefaultStatus();
+
+        if (provider.setStatusForRole) {
+            return provider.setStatusForRole(role, defaultStatus);
+        } else {
+            this.#userAPI.error("User provider does not support resetting role status");
+        }
+    }
+
+    /**
+     * Resets the status of all operators to their default status
+     * @returns {Promise<Boolean>} true if operation was successful, otherwise false.
+     */
+    async resetAllStatuses() {
+        const allStatusRoles = await this.getAllStatusRoles();
+
+        return Promise.all(allStatusRoles.map(role => this.resetStatusForRole(role)));
+    }
+
+    /**
+     * The default status. This is the status that will be used before the user has selected any status.
+     * @returns {Promise<Status>} the default operator status if no other has been set.
+     */
+    async getDefaultStatus() {
+        const allStatuses = await this.getPossibleStatuses();
+
+        return allStatuses[0];
+    }
+
+    /**
+     * All possible status roles. A status role is a user role that can provide status. In some systems
+     * this may be all user roles, but there may be cases where some users are not are not polled
+     * for status if they do not have a real-time operational role.
+     *
+     * @returns {Promise<Array<import("./UserAPI").Role>>} the default operator status if no other has been set.
```

Comment:
It looks ok to me and seems to be working ok? Maybe I fixed it since you left this comment and I've just forgotten though :) Let me know if there's still an issue.